src/context/ApiContext.tsx

```diff
@@ -0,0 +1,139 @@
+import { useQuery } from '@tanstack/react-query'
+import { useSession } from 'next-auth/react'
+import getConfig from 'next/config'
+import { createContext, useContext, useEffect, useRef, useState } from 'react'
+
+import { Colour, Image } from 'src/components/Types'
+import { useGuildContext } from 'src/context/GuildContext'
+import { ColourToByte } from 'src/helpers/Colours'
+
+type ApiContextType =
+  | {
+      image?: Image
+      setPixel?(x: number, y: number, colour: Colour): Promise<void>
+    }
+  | undefined
+
+const ApiContext = createContext<ApiContextType>(undefined)
+
+export function ApiContextProvider({
+  children
+}: {
+  children: React.ReactNode
+}) {
+  const { guildId } = useGuildContext()
+  const session = useSession()!
+
+  const webSocket = useRef<WebSocket>()
+
+  const [imageWidth, setImageWidth] = useState<number>()
+  const [imageHeight, setImageHeight] = useState<number>()
+  const [imageData, setImageData] = useState<Uint8Array>()
+
+  useEffect(() => {
+    if (!guildId) return
+
+    const {
+      publicRuntimeConfig: { WS_URL }
+    } = getConfig()
+
+    const ws = new WebSocket(`${WS_URL}servers/${guildId}/ws`)
+
+    ws.onopen = () => {
+      ws.send(session.data?.accessToken!)
+    }
+
+    ws.onmessage = async (event) => {
+      const blob = event.data as Blob
+      const buffer = await blob.arrayBuffer()
+
+      const bytes = new Uint8Array(buffer)
+      const x = bytes[0] * 255 + bytes[1]
```

Comment:
Shifting would be more descriptive of what's actually happening here IMO
Also I think it should be 256 instead of 255 right?
```suggestion
      const x = bytes[0] << 8 + bytes[1]
```